pkg/metacni/metacni.go

```diff
@@ -207,14 +210,54 @@ func popDevice(devicePool string, allocatedDevices map[string]*[]string)(string,
   return device, nil
 }
 
+func setIpv6Sysctls(args *cniArgs) error {
+  runtime.LockOSThread()
+  defer runtime.UnlockOSThread()
+  // save the current namespace
+  origNs, err := ns.GetCurrentNS()
+  if err != nil {
+    return errors.New("failed to get current network namespace due to:" + err.Error())
+  }
+  // enter to the Pod's network namespace
+  podNs, err := ns.GetNS(args.netns)
+  if err != nil {
+    return errors.New("failed to get Pod's network namespace due to:" + err.Error())
+  }
+  err = podNs.Set()
+  if err != nil {
+    return errors.New("failed to switch to Pod's network namespace due to:" + err.Error())
+  }
+  defer func() {
+    podNs.Close()
+    origNs.Set()
+  }()
+  // set sysctls
+  log.Println("IPv6 sysctl: net.ipv6.conf.all.disable_ipv6 = 0")
+  _, err = sysctl.Sysctl("net.ipv6.conf.all.disable_ipv6", "0")
+  if err != nil {
+    return errors.New("failed to set 'disable_ipv6' sysctl due to:" + err.Error())
+  }
+  return nil
+}
+
 func setupNetworking(args *cniArgs) (*current.Result, error) {
   danmClient, err := createDanmClient(args.stdIn)
   if err != nil {
     return nil, errors.New("failed to create DanmClient due to:" + err.Error())
   }
+  // set IPv6 related sysctls, if at least one IPv6 CIDR is set on any interfaces
+  for _, nicParams := range args.interfaces {
+    if nicParams.Ip6 != "" {
```

Comment:
just because somebody asked for IPv6 from Pod Spec, it doesn't automatically mean the network he is connecting to is also IPv6 enabled
presence of Net6 should be the indicator we need to make this setting

Reply:
BTW we are already looping over the interfaces map a couple of lines later.
combining with the other comment asking for per interface  setting, you can just put this could into the scope of the existing loop

Reply:
code has been refactored heavily...